src/core/execute.c

```diff
@@ -5322,6 +5323,7 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
                 "%sProtectProc: %s\n"
                 "%sProcSubset: %s\n",
                 prefix, c->umask,
+                prefix, empty_to_root(c->binary_directory),
```

Comment:
Hmm? `empty_to_root` doesn't make much sense to me here, if `binary_directory` is empty your code doesn't treat it the same as `BinaryDirectory=/`. You probably want `strempty`.

Reply:
I believe if it is empty ExecVe will go with the absolute path, which should be equivalent to ```BinaryDirectory=/```.

Reply:
That's just my understanding though

Reply:
But that's not the same, though. If it's empty and you put `foo`, you are never going to be allowed to execute `/foo`.

Reply:
Ok sounds good, I'll just use strempty. Thanks!